include/cxxopts.hpp

```diff
@@ -2111,36 +2169,34 @@ OptionAdder::operator()
   std::string arg_help
 )
 {
-  std::string short_sw, long_sw;
-  std::tie(short_sw, long_sw) = values::parser_tool::SplitSwitchDef(opts);
-
-  if (!short_sw.length() && !long_sw.length())
-  {
-    throw_or_mimic<exceptions::invalid_option_format>(opts);
+  OptionNames all_option_names = values::parser_tool::split_option_names(opts);
+    // Note: All names will be non-empty!
+  std::string short_sw {""};
+  OptionNames long_option_names = all_option_names;
```

Comment:
do you need this extra copy? It looks like you don't use `all_option_names` again.

Reply:
I don't. I just wanted to clarify this is no longer `all_option_names`.  Ideally, a compiler could optimize this away; but I suppose I could use `std::move`. On the other hand, there are soooo many string copies in cxxopts - it could really stand to use `string_view` everywhere.